x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransformNodes.java

```diff
@@ -147,4 +153,19 @@ public static void warnIfNoTransformNodes(ClusterState clusterState) {
             HeaderWarning.addWarning(TransformMessages.REST_WARN_NO_TRANSFORM_NODES);
         }
     }
+
+    /**
+     * Select any node among provided nodes that satisfies all of the following:
+     *  - is a transform node
+     *  - is a remote cluster client node
```

Comment:
This isn't strictly true given the implementation of this method.  It's only true because this method only gets called after determining that a remote cluster node is required.

`nodeCanRunThisTransform()` only enforces remote cluster client nodes if the transform requires one.